configs/components/cmake.rb

```diff
@@ -1,15 +1,19 @@
 component "cmake" do |pkg, settings, platform|
   # Source-Related Metadata
-  pkg.version "3.2.3"
-  pkg.md5sum "d51c92bf66b1e9d4fe2b7aaedd51377c"
-  pkg.url "https://cmake.org/files/v3.2/#{pkg.get_name}-#{pkg.get_version}.tar.gz"
+  if platform.is_solaris?
+    pkg.version "3.12.4"
+    pkg.url "https://cmake.org/files/v3.12/#{pkg.get_name}-#{pkg.get_version}.tar.gz"
+    pkg.md5sum "4970869480ee37c8f4c8e7f83c4359cd"
+  else
+    pkg.version "3.2.3"
+    pkg.url "https://cmake.org/files/v3.2/#{pkg.get_name}-#{pkg.get_version}.tar.gz"
+    pkg.md5sum "d51c92bf66b1e9d4fe2b7aaedd51377c"
+  end
 
   pkg.mirror "#{settings[:buildsources_url]}/#{pkg.get_name}-#{pkg.get_version}.tar.gz"
 
   if platform.name =~ /sles-12/
     pkg.apply_patch 'resources/patches/cmake/avoid-select-sles-12.patch'
-  elsif platform.is_solaris?
-    pkg.apply_patch 'resources/patches/cmake/use-g++-as-linker-solaris.patch'
```

Comment:
Do we have any guidelines around deleting patch files that have (potentially) become obsolete?

Reply:
Not that I know of, but this patch only applies to the older cmake package on solaris, which (with these changes) no longer exists, so I don't think it makes sense to keep it.

Reply:
(I've removed that patch file now, too)